src/commons/api/roas.rs

```diff
@@ -280,22 +286,186 @@ impl Ord for RoaDefinition {
+        RoaConfiguration { payload, comment }
+    }
+
+    pub fn unpack(self) -> (RoaPayload, Option<String>) {
+        (self.payload, self.comment)
+    }
+
+    pub fn payload(&self) -> RoaPayload {
+        self.payload
+    }
+
+    pub fn comment(&self) -> Option<&String> {
+        self.comment.as_ref()
+    }
+
+    /// Ensures that the payload uses an explicit max length
+    pub fn into_explicit_max_length(self) -> Self {
+        RoaConfiguration {
+            payload: self.payload.into_explicit_max_length(),
+            comment: self.comment,
+        }
+    }
+}
+
+impl fmt::Display for RoaConfiguration {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.payload)?;
+        if let Some(comment) = &self.comment {
+            write!(f, " # {}", comment)?;
+        }
+        Ok(())
+    }
+}
+
+impl FromStr for RoaConfiguration {
+    type Err = AuthorizationFmtError;
+
+    // "192.168.0.0/16 => 64496"
+    // "192.168.0.0/16 => 64496 # my nice ROA"
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut parts = s.splitn(2, '#');
+        let payload_part = parts.next().ok_or_else(|| AuthorizationFmtError::auth(s))?;
+
+        let payload = RoaPayload::from_str(payload_part)?;
+        let comment = parts.next().map(|s| s.trim().to_string());
+
+        Ok(RoaConfiguration { payload, comment })
+    }
+}
+
+impl Ord for RoaConfiguration {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.payload.cmp(&other.payload)
+    }
+}
+
+impl PartialOrd for RoaConfiguration {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl From<RoaPayload> for RoaConfiguration {
+    fn from(payload: RoaPayload) -> Self {
+        RoaConfiguration { payload, comment: None }
+    }
+}
+
+//------------ ConfiguredRoa -----------------------------------------------
+
+/// Defines an existing ROA configuration and related information.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct ConfiguredRoa {
+    #[serde(flatten)]
+    roa_configuration: RoaConfiguration,
+    // roa_objects: Vec<RoaInfo>, // will be added when #864 is implemented
+}
+
+impl ConfiguredRoa {
+    pub fn new(roa_configuration: RoaConfiguration) -> Self {
+        ConfiguredRoa { roa_configuration }
+    }
+
+    pub fn payload(&self) -> RoaPayload {
+        self.roa_configuration.payload
+    }
+
+    pub fn asn(&self) -> AsNumber {
+        self.roa_configuration.payload.asn()
+    }
 
-impl fmt::Display for RoaDefinitions {
+    pub fn prefix(&self) -> TypedPrefix {
+        self.roa_configuration.payload.prefix()
+    }
+
+    pub fn effective_max_length(&self) -> u8 {
+        self.roa_configuration.payload.effective_max_length()
+    }
+
+    pub fn nr_of_specific_prefixes(&self) -> u128 {
+        self.roa_configuration.payload.nr_of_specific_prefixes()
+    }
+
+    pub fn as_roa_ip_address(&self) -> RoaIpAddress {
```

Comment:
Couldn't this be done with a `From`/`Into` impl?

Reply:
It could, but because of how references and `Deref` is used here that would result in less readable code. So I prefer to keep it as is.